test/reduce.jl

```diff
@@ -172,6 +175,19 @@ for f in (sum3, sum4, sum7, sum8)
 end
 @test typeof(sum(Int8[])) == typeof(sum(Int8[1])) == typeof(sum(Int8[1 7]))
 
+@testset "`sum` of empty collections with `init`" begin
+    @testset for init in [0, 0.0]
+        @test sum([]; init = init) === init
+        @test sum((x for x in [123] if false); init = init) === init
+        @test sum(nothing, []; init = init) === init
```

Comment:
Is passing `nothing` as the first argument documented?

Reply:
I added a comment about it:

https://github.com/JuliaLang/julia/blob/6918213ef2081430e7b6f586b1013aee6641523e/test/reduce.jl#L189-L191

I can instead do

```julia
@testset "`sum` of empty collections with `init`" begin
    function noncallable end
    @testset for init in [0, 0.0]
        # ...
        @test sum(noncallable, []; init = init) === init
        # ...
    end
end

```

if that's better.

Reply:
Actually, now I think `function noncallable end` is better. (But we can rollback if you want.)